### PR TITLE
MAINT: Bump to 3.10 min req

### PR DESCRIPTION
**Pull request by larsoner**
Suggested move to require 3.10+. Clean up something that could have been cleaned up once 3.9+ was required. Clean up incorrect installation versions for stable and prev-stable (which used to be the same!).

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -115,38 +115,26 @@ jobs:
...
+    # ubuntu / 3.10 / mne-prev / full / validator-stable
     # ubuntu / 3.12 / mne-stable / minimal / validator-stable
     timeout-minutes: 60
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: ["3.9", "3.12"]  # Oldest and newest supported versions
+        os: [ubuntu-latest, macos-latest, macos-13, windows-latest]
```
Comment (larsoner) [inline .github/workflows/unit_tests.yml:130]: Based on the comments below I think this was also incorrectly specified once `macos-latest` switched from `-13` (x86_64) to `-14` (arm64). In any case might as well test both now I think

**Review by sappelhoff — APPROVED** [review]
LGTM! Thanks Eric.

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -115,38 +115,26 @@ jobs:
     #
     # 6 checks:
     # for each OS (ubuntu, macos, windows):
```
Comment (sappelhoff) [inline .github/workflows/unit_tests.yml:117]:
```suggestion
    # for each OS (ubuntu, macos, macos-13, windows):
    # (NOTE: macos-13=x86_64, macos>13=arm64)
```

File: .github/workflows/unit_tests.yml [hunk 3]
```diff
@@ -115,38 +115,26 @@ jobs:
...
-    # 3.9 / mne-stable / full / validator-stable
-    # 3.12 / mne-stable / full / validator-stable
-    #
-    # 1 additional check for Apple Silicon (doesn't support Python 3.9):
+    # 3.10 / mne-stable / full / validator-stable
     # 3.12 / mne-stable / full / validator-stable
     #
     # 5 additional checks with alternative MNE-Python and BIDS validator versions:
```
Comment (sappelhoff) [inline .github/workflows/unit_tests.yml:121]:
```suggestion
    # 3 additional checks with alternative MNE-Python and BIDS validator versions:
```

**Review by sappelhoff — COMMENTED** [review]
I'll adjust the branch protection rules for required checks.

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (sappelhoff) [reply .github/workflows/unit_tests.yml:117]: actually, the `    # 6 checks:` above needs to be changed to "8 checks" 🤔 

and it'd be nice to have a changelog entry saying that we now require Python 3.10 minimum. Could you add that please @larsoner ? If not, let me know and I'll do it at some later point. Thanks again.

**Comment by larsoner** [comment]
Okay should be good @sappelhoff !

**Pull request merged**